### PR TITLE
feat(user model): add grandchildren association for users

### PR DESCRIPTION
**Pull request by LeeRenJie**
User model
```ruby
class User < ApplicationRecord
  has_many :children,
  -> (user) { unscope(:where).where("father_id = :id OR mother_id = :id", id: user.id) },
   class_name: "User"
   has_many :grandchildren,
  -> (user) { unscope(:where).where("father_id IN (:ids) OR mother_id IN (:ids)", ids: user.children.ids) },
  class_name: "User"
  belongs_to :mother, class_name: "User", optional: true
  belongs_to :father, class_name: "User", optional: true
end
```

Output
image

### CODE REVIEW

**Review by tlkiong — CHANGES_REQUESTED** [review]
make this PR smaller if you want me review

File: test-app/app/models/user.rb [hunk 1]
```diff
@@ -1,9 +1,16 @@
 class User < ApplicationRecord
   has_one :articles
-  has_many :children, ->(user) { unscope(:where).where("father_id = :id OR mother_id = :id", id: user.id) }, class_name: "User"
+
+  has_many :children,
+  -> (user) { unscope(:where).where("father_id = :id OR mother_id = :id", id: user.id) },
+   class_name: "User"
+
+   has_many :grandchildren,
```
Comment (tlkiong) [inline test-app/app/models/user.rb:8]: Interesting. 
Can you make it into 1 sql query?

Comment (LeeRenJie) [reply test-app/app/models/user.rb:8]: Can try, abit hard 😂

Comment (tlkiong) [reply test-app/app/models/user.rb:8]: wow. a bit only? ok. ganbatte

Comment (LeeRenJie) [reply test-app/app/models/user.rb:8]: alot hard 😂

Comment (tlkiong) [reply test-app/app/models/user.rb:8]: still, ganbatte.
:P

**Review by tlkiong — APPROVED** [review]

**Pull request merged**